spec/unit/rust-crypto/verification.spec.ts

```diff
@@ -152,3 +405,148 @@ function makeMockedInner(): Mocked<RustSdkCryptoJs.VerificationRequest> {
         },
     } as unknown as Mocked<RustSdkCryptoJs.VerificationRequest>;
 }
+
+interface CrossSigningKeys {
+    master_key: any;
+    self_signing_key: any;
+    user_signing_key: any;
+}
+
+/** create an Olm machine and device/cross-signing keys for a user */
+async function initOlmMachineAndKeys(
+    userId: string,
+    deviceId: string,
+): Promise<[RustSdkCryptoJs.OlmMachine, IDeviceKeys, CrossSigningKeys]> {
+    const olmMachine = await RustSdkCryptoJs.OlmMachine.initialize(
+        new RustSdkCryptoJs.UserId(userId),
+        new RustSdkCryptoJs.DeviceId(deviceId),
+        undefined,
+        undefined,
+    );
+    const { uploadKeysRequest, uploadSignaturesRequest, uploadSigningKeysRequest } =
+        await olmMachine.bootstrapCrossSigning(true);
+    const deviceKeys = JSON.parse(uploadKeysRequest.body).device_keys;
+    await olmMachine.markRequestAsSent(
+        uploadKeysRequest.id,
+        uploadKeysRequest.type,
+        '{"one_time_key_counts":{"signed_curve25519":100}}',
+    );
+    const crossSigningSignatures = JSON.parse(uploadSignaturesRequest.body);
+    if (crossSigningSignatures[userId] && crossSigningSignatures[userId][deviceId]) {
```

Comment:
it looks like this should always be true? I'd be tempted to get rid of this condition.

Alternatively, a ? operator would simplify:

```suggestion
    if (crossSigningSignatures[userId]?.[deviceId]) {
```

Doesn't matter though.